core/domain/suggestion_services.py

```diff
@@ -0,0 +1,382 @@
+# be removed.
+def get_thread_id_from_suggestion_id(suggestion_id):
+    """Gets the thread_id from the suggestion_id.
+
+    Args:
+        suggestion_id: str. The ID of the suggestion.
+
+    Returns:
+        str. The thread ID linked to the suggestion.
+    """
+    return suggestion_id[suggestion_id.find('.') + 1:]
+
+
+def is_suggestion_handled(suggestion):
+    """Checks if the suggestion has been accepted or rejected.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            checked.
+
+    Returns:
+        bool. Whether the suggestion has been handled or not.
+    """
+    return (suggestion.status not in [suggestion_models.STATUS_IN_REVIEW,
+                                      suggestion_models.STATUS_RECEIVED])
+
+
+def get_commit_message_for_suggestion(author_username, commit_message):
+    """Returns a modified commit message for an accepted suggestion.
+
+    Args:
+        author_username: str. Username of the suggestion author.
+        commit_message: str. The original commit message submitted by the
+            suggestion author.
+
+    Returns:
+        str. The modified commit message to be used in the exploration commit
+        logs.
+    """
+    return '%s %s: %s' % (
+        feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX,
+        author_username, commit_message)
+
+
+def accept_suggestion(suggestion, reviewer_id, commit_message):
+    """Accepts the given suggestion after validating it.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            accepted.
+        reviewer_id: str. The ID of the reviewer accepting the suggestion.
+        commit_message: str. The commit message.
+
+    Raises:
+        Exception: The suggestion is already handled.
+        Exception: The suggestion is not valid.
+        Exception: The commit message is empty.
+    """
+    if is_suggestion_handled(suggestion):
+        raise Exception('The suggestion has already been accepted/rejected.')
+    if not commit_message or not commit_message.strip():
+        raise Exception('Commit message cannot be empty.')
+
+    try:
+        # If an exception is raised, the suggestion status will be set as
+        # STATUS_INVALID.
+        suggestion.validate()
+    except utils.ValidationError:
+        invalidate_suggestion(suggestion)
+        raise Exception('The given suggestion is not valid.')
+
+    author_name = user_services.get_username(suggestion.author_id)
+    commit_message = get_commit_message_for_suggestion(
+        author_name, commit_message)
+    mark_review_completed(
+        suggestion, suggestion_models.STATUS_ACCEPTED, reviewer_id)
+    suggestion.accept(commit_message)
+
+    feedback_services.create_message(
+        get_thread_id_from_suggestion_id(suggestion.suggestion_id), reviewer_id,
+        feedback_models.STATUS_CHOICES_FIXED, None,
+        'Accepted by %s' % reviewer_id)
```

Comment:
Don't we already have a default thing show up when the status of a thread changes? Maybe the message should be an optional addition by the reviewer?

Reply:
Yeah. The message shows up as User_name <gap> Status changed to 'Fixed' <next line> (message). So I think by default the message can be blank and if the reviewer wants to add something else then we can pass it here. Also with this view I think the bot isn't necessary because the way it shows up makes sense (The user did actually change the status and optionally give the message).

Reply:
Passes review_message argument. If None, only the default message will be shown. If something is passed, that will be shown in the thread.